runtime/libcn/include/cn-executable/eval.h

```diff
@@ -0,0 +1,426 @@
+#ifndef CN_EVAL_H
+#define CN_EVAL_H
+
+#include "bump_alloc.h"
+#include "hash_table.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/* Wrappers for C types */
+
+/* Signed bitvectors */
+
+typedef struct cn_bits_i8 {
```

Comment:
I'd probably prefer there to be a different header for the type definitions (e.g. `fulminate/ghost_types.h`), which gets included in this header, and then the rest can make up `eval.h`